tests/unit/lib/charms/vault_k8s/v0/test_vault_autounseal.py

```diff
@@ -118,41 +118,41 @@ def test_given_unit_is_leader_when_set_autounseal_data_then_relation_data_is_upd
             relations=[vault_autounseal_relation],
             leader=True,
         )
-        action = scenario.Action(
-            "set-autounseal-data",
-            params={
-                "ca-certificate": "my ca certificate",
-                "relation-id": str(vault_autounseal_relation.relation_id),
-                "vault-address": "https://vault.example.com",
-                "mount-path": "charm-autounseal",
-                "key-name": "some key name",
-                "approle-role-id": "some approle id",
-                "approle-secret-id": "some approle secret id",
-            },
-        )
-
-        action_output = self.ctx.run_action(action, state_in)
 
-        assert action_output.success is True
-        assert (
-            action_output.state.relations[0].local_app_data["ca_certificate"]
-            == "my ca certificate"
-        )
-        assert (
-            action_output.state.relations[0].local_app_data["address"]
-            == "https://vault.example.com"
-        )
-        assert action_output.state.relations[0].local_app_data["mount_path"] == "charm-autounseal"
-        assert action_output.state.relations[0].local_app_data["key_name"] == "some key name"
-        assert "credentials_secret_id" in action_output.state.relations[0].local_app_data
-        credentials_secret_id = action_output.state.relations[0].local_app_data[
-            "credentials_secret_id"
-        ]
-        juju_secret = action_output.state.secrets[0]
-        assert juju_secret.contents == {
-            0: {"role-id": "some approle id", "secret-id": "some approle secret id"}
-        }
-        assert juju_secret.id == credentials_secret_id
+        state_out = self.ctx.run(
+            self.ctx.on.action(
+                "set-autounseal-data",
+                params={
+                    "ca-certificate": "my ca certificate",
+                    "relation-id": str(vault_autounseal_relation.id),
+                    "vault-address": "https://vault.example.com",
+                    "mount-path": "charm-autounseal",
+                    "key-name": "some key name",
+                    "approle-role-id": "some approle id",
+                    "approle-secret-id": "some approle secret id",
+                },
+            ),
+            state_in,
+        )
+
+        relation_data = state_out.get_relation(vault_autounseal_relation.id).local_app_data
+        assert relation_data["ca_certificate"] == "my ca certificate"
+        assert relation_data["address"] == "https://vault.example.com"
+        assert relation_data["address"] == "https://vault.example.com"
+        assert relation_data["mount_path"] == "charm-autounseal"
+        assert relation_data["key_name"] == "some key name"
```

Comment:
Just a nit but do we need individual assertions instead of asserting the complete dict? The dict is set as a whole I believe so I don't really see a value in separating them

Reply:
Done